core/services/keeper/orm.go

```diff
@@ -132,6 +161,17 @@ ORDER BY upkeep_registrations.id ASC, upkeep_registrations.upkeep_id ASC
 	return upkeeps, err
 }
 
+// binaryForTurn gets the previous turns starting block hash and converts it to binary
+// the finality depth is subtracted to avoid any contention that could happen
+func binaryForTurn(blockNumber int64, registry Registry, head *types.Head, finality uint32) string {
+	lookbackBlock := blockNumber - (blockNumber % int64(registry.BlockCountPerTurn)) - int64(finality)
+	hashAtHeight := head.HashAtHeight(lookbackBlock)
+	bigInt := new(big.Int)
+	bigInt.SetString(hashAtHeight.Hex(), 0)
+	binaryString := fmt.Sprintf("%b", bigInt)
+	return binaryString
```

Comment:
just wondering... does this have to be a string? can postgres not handle raw `[]byte`?

Reply:
it errors out on the conversion to bits when i tried